sklearn/ensemble/_base.py

```diff
@@ -129,23 +152,54 @@ def __init__(self, base_estimator, *, n_estimators=10, estimator_params=tuple())
     def _validate_estimator(self, default=None):
         """Check the base estimator.
 
-        Sets the base_estimator_` attributes.
+        Sets the `estimator_` attributes.
         """
-        if self.base_estimator is not None:
-            self.base_estimator_ = self.base_estimator
+        if self.estimator is not None and (
+            self.base_estimator not in [None, "deprecated"]
+        ):
+            raise ValueError(
+                "Both `estimator` and `base_estimator` were set. Only set `estimator`."
+            )
+
+        if self.estimator is not None:
+            self._estimator = self.estimator
+        elif self.base_estimator not in [None, "deprecated"]:
+            warnings.warn(
+                "`base_estimator` was renamed to `estimator` in version 1.2 and "
+                "will be removed in 1.4.",
+                FutureWarning,
+            )
+            self._estimator = self.base_estimator
         else:
-            self.base_estimator_ = default
+            self._estimator = default
+
+        if self._estimator is None:
```

Comment:
At this point, self._estimator can't be None. You can remove this since it's unreachable code